Joe11122/sub nav (#403)
Esri/calcite-react: src/SubNav/doc/SubNav.mdx

SubNav provides a second level of hierarchy to a site. While TopNav includes the
top-most level of navigation, SubNav includes navigation for sections nested
inside that. SubNav has a tabbed navigation pattern and a title. 

### ⛔️ Use of `legacy` prop ⛔️

> The `legacy` prop must be included to keep legacy styles from before `v0.55.0` releae. 
> SubNavActions may only be used with legacy. Please take a look at the SubNav props 
> table for a full list of props that are only accepted with legacy.

Reviewer: ```suggestion
The `legacy` prop must be included to keep original styles from before the `v0.55.0` release. 
`SubNavActions` may only be used with legacy. Please refer to the `SubNav` props 
table for a full list of props that are only accepted with `legacy`.
```

  </SubNav>
</Playground>

## With Breadcrumbs

<Playground>
  <SubNav>
    <SubNavTitle>
      <Breadcrumbs>
        <Crumb href="#">Thing</Crumb>
        <Crumb>Thing</Crumb>
        <Crumb href="#">Thing</Crumb>
        <Crumb href="#">Current</Crumb>
      </Breadcrumbs>
    </SubNavTitle>
    <SubNavList>
      <SubNavLink active href="#">
        Glens
      </SubNavLink>
      <SubNavLink href="#">Dales</SubNavLink>
      <SubNavLink href="#">Meadows</SubNavLink>
    </SubNavList>
  </SubNav>
</Playground>

## Different Background Colors

> Using either a calcite-color (ex. 'purple') or a custom color.

Reviewer: ```suggestion
`backgroundColor` can take a string to reference a color value from the `CalciteThemeProvider` palette object. Alternatively you may provide a `hex`, `rgb` or any other valid css color definition.
````

            Glens
          </SubNavLink>
          <SubNavLink href="#">Dales</SubNavLink>
          <SubNavLink href="#">Meadows</SubNavLink>
        </SubNavList>
      </SubNav>
    </GuideExample>

    <GuideExample label="with actions">
      <SubNav legacy={true}>
        <SubNavTitle href="#">Fields</SubNavTitle>
        <SubNavList>
          <SubNavLink active href="#">
            Glens
          </SubNavLink>
          <SubNavLink href="#">Dales</SubNavLink>
          <SubNavLink href="#">Meadows</SubNavLink>
        </SubNavList>
        <SubNavActions>
          <Button white small>
            New Field
          </Button>
        </SubNavActions>
      </SubNav>
    </GuideExample>

    <GuideExample label="no nav - with actions">
      <SubNav legacy={true}>
        <SubNavTitle href="#">Fields</SubNavTitle>
        <SubNavActions>
          <Button white small>
            New Field
          </Button>
        </SubNavActions>
      </SubNav>
    </GuideExample>

    <GuideExample label="backgroundImage">
      <SubNav
        backgroundImage={greenBlueBg}
        gradientFromColor="#70be49"
        gradientToColor="#1688aa"
        overlayGradient={false}

>         legacy={true}

Reviewer: ```suggestion
        legacy
```

        backgroundImage={grayCityBg}
        gradientFromColor="#70BB51"
        gradientToColor="#2088A7"

>         legacy={true}

Reviewer: ```suggestion
        legacy
```
